CRM/Contact/Form/Task/Email.php

```
@@ -37,10 +37,14 @@ public function preProcess() {
 
     $cid = CRM_Utils_Request::retrieve('cid', 'String', $this, FALSE);
 
-    // Allow request to specify email id rather than contact id
+    // Allow request to specify email id or address rather than contact id
     $toEmailId = CRM_Utils_Request::retrieve('email_id', 'String', $this);
-    if ($toEmailId) {
-      $toEmail = civicrm_api('email', 'getsingle', ['version' => 3, 'id' => $toEmailId]);
+    $toEmailAddress = CRM_Utils_Request::retrieve('email', 'String', $this);
+    if ($toEmailId || $toEmailAddress) {
+      $filter = $toEmailId
+        ? ['id' => $toEmailId]
+        : ['email' => $toEmailAddress];
+      $toEmail = civicrm_api('email', 'getsingle', array_merge(['version' => 3], $filter));
```

@reneolivo this would fail hard if that email exists more than once in the db -